src/frontend/src/expr/type_inference/func.rs

```diff
@@ -523,6 +523,27 @@ fn infer_type_for_special(
                 .into()),
             }
         }
+        ExprType::ArrayDistinct => {
+            ensure_arity!("array_distinct", | inputs | == 1);
+            let ret_type = inputs[0].return_type();
+            if inputs[0].is_unknown() {
+                return Err(ErrorCode::BindError(
+                    "could not determine polymorphic type because input has type unknown"
+                        .to_string(),
+                )
+                .into());
+            }
+            match ret_type {
+                DataType::List {
+                    datatype: list_elem_type,
+                } => Ok(Some(DataType::List {
+                    datatype: list_elem_type,
+                })),
+                _ => Err(
+                    ErrorCode::BindError("Array contains different value types".to_string()).into(),
+                ),
```

Comment:
This error corresponds to `select array_distinct(1);` when the input is not an array. Use a more proper error message or return `Ok(None)` to get the general "builtin function not found/implemented" message.